### PR TITLE
try injecting from platform chart values

### PR DESCRIPTION
**Pull request by jklugherz**

### CODE REVIEW

File: charts/pipeline-runner/templates/environment.yaml [hunk 1]
```diff
@@ -8,3 +8,4 @@ data:
   {{- with .Values.environment }}
     {{- toYaml . | nindent 2 }}
   {{- end }}
+  SEQR_APP_HAIL_SEARCH_DATA_DIR: {{ .Values.global.HAIL_SEARCH_DATA_DIR | quote }}
```
[the remark this answers is not in the document]
Comment (bpblanken) [reply charts/pipeline-runner/templates/environment.yaml:11]: Also, I think we wanted this set iff the app `HAIL_SEARCH_DATA_DIR ` and the pipeline `HAIL_SEARCH_DATA_DIR` were different right?  This strategy gets us most but not all the way there I think?

[the remark this answers is not in the document]
Comment (bpblanken) [reply charts/pipeline-runner/templates/environment.yaml:11]: Maybe if we injected as `.Values.pipeline-runner.HAIL_SEARCH...` and `.Values.seqr.HAIL_SEARCH`?  Then this snippet could compare the two and set `SEQR_APP_HAIL_SEARCH_DATA_DIR` accordingly?

[the remark this answers is not in the document]
Comment (jklugherz) [reply charts/pipeline-runner/templates/environment.yaml:11]: there's no way to compare sibling chart values directly but I was able to export the `hail-search` variable as `SEQR_APP_HAIL_SEARCH_DATA_DIR` and access it via .`Values.globals` inside of `charts/pipeline-runner/templates/environment.yaml`

(I found the solution [here](https://stackoverflow.com/questions/70506589/how-to-use-parameters-of-one-subchart-in-another-subchart-in-helm))

File: charts/hail-search/values.yaml [hunk 2]
```diff
@@ -54,3 +54,8 @@ tolerations: []
 
 global:
   seqrPlatformDeploy: false
+
+exports:
```
Comment (bpblanken) [inline charts/hail-search/values.yaml:58]: Don't we want to export the previously set
```
environment:
  HAIL_SEARCH_DATA_DIR: ...
```

here?  So if a user, in the `seqr-platform` specifies a different directory for this:

```
hail-search:
  environment:
     HAIL_SEARCH_DATA_DIR: ...
```

that change would propagate through to the export?

Comment (bpblanken) [reply charts/hail-search/values.yaml:58]: Oh wait... we can't interpolate super well in this file.

File: charts/hail-search/values.yaml [hunk 3]
```diff
@@ -54,3 +54,8 @@ tolerations: []
...
+  environment:
+    global:
```
Comment (bpblanken) [inline charts/hail-search/values.yaml:60]: We also might be able to just export this as as `hail-search` value:

exports:
  hail-search:
    environment:
      ...
```

File: charts/hail-search/values.yaml (hunk 2 above)
Comment (bpblanken) [reply charts/hail-search/values.yaml:58]: None of these comments are helpful 🤣 ... but I think we probably need approach this different.  Let's set up time to talk through this!

File: charts/pipeline-runner/templates/deployment.yaml [hunk 4]
```diff
@@ -81,6 +81,9 @@ spec:
           envFrom:
             - configMapRef:
                 name: {{ $.Chart.Name }}
+            - configMapRef:
+                name: seqr-platform
+                optional: true
```
Comment (bpblanken) [inline charts/pipeline-runner/templates/deployment.yaml:86]: yay!  this is awesome.

**Review by bpblanken — APPROVED** [review]

**Pull request merged**